pydantic/_internal/_generate_schema.py

```diff
@@ -905,6 +907,13 @@ def _pattern_schema(self, pattern_type: Any) -> core_schema.CoreSchema:
         else:
             raise PydanticSchemaGenerationError(f'Unable to generate pydantic-core schema for {pattern_type!r}.')
 
+    def _hashable_schema(self) -> core_schema.CoreSchema:
+        return core_schema.custom_error_schema(
+            core_schema.is_instance_schema(collections.abc.Hashable),
+            custom_error_type='is_hashable',
+            custom_error_message='Input should be hashable',
+        )
```

Comment:
This adds an error_type, but it's very similar to the `_type_schema` and I think it's reasonable to have a special message for this. I think it looks better than the error message we had for arbitrary types before. (Which you can see in the updated tests below.)

(Also, thanks github copilot, wouldn't have thought to use `custom_error_schema` without it.)

Reply:
Also @tpdorsey for reference — not sure if we should do something to make sure the custom_error_type is tracked better/make sure it gets documented.